### PR TITLE
[Importer] supporting large file(10MB) for local importer

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- supporting large file(10MB) for local importer

## How was this patch tested?

- Manually tested in Mysql, Oracle, and Postgres DBs

### CODE REVIEW

File: desktop/core/src/desktop/conf.py [hunk 1]
```diff
@@ -642,13 +642,16 @@ def default_from_email():
...
+from indexer.conf import ENABLE_DIRECT_UPLOAD  # Circular dependency 
 
 def default_database_options():
   """Database type dependent options"""
   if DATABASE.ENGINE.get().endswith('oracle'):
     return {'threaded': True}
   elif DATABASE.ENGINE.get().endswith('sqlite3'):
     return {'timeout': 30}
+  elif DATABASE.ENGINE.get().endswith('mysql') and ENABLE_DIRECT_UPLOAD:
+    return {'init_command': 'SET GLOBAL max_allowed_packet=67108864'}
```
Comment (Harshg999) [inline desktop/core/src/desktop/conf.py:654]: Nit: Can we add a comment for what this number denotes?

File: desktop/core/src/desktop/settings.py [hunk 2]
```diff
@@ -117,6 +118,8 @@
 # Examples: "http://media.lawrence.com/media/", "http://example.com/media/"
 MEDIA_URL = ''
 
+if ENABLE_DIRECT_UPLOAD:
+  DATA_UPLOAD_MAX_MEMORY_SIZE = 67108864
```
Comment (Harshg999) [inline desktop/core/src/desktop/settings.py:122]: Nit: Same, to add a comment for this number

File: desktop/libs/indexer/src/indexer/indexers/sql.py [hunk 3]
```diff
@@ -355,23 +357,30 @@ def create_table_from_local_file(self, source, destination, start_time=-1):
           _csv_rows.append(tuple(row))
 
         if _csv_rows:
-          csv_rows = str(_csv_rows)[1:-1]
 
-          if dialect in ('hive', 'mysql'):
-            sql += '''\nINSERT INTO %(database)s.%(table_name)s VALUES %(csv_rows)s;\n'''% {
+          insert_sql = ""
+          for count in range(1 + len(_csv_rows)//5000):
```
Comment (Harshg999) [inline desktop/libs/indexer/src/indexer/indexers/sql.py:362]: Nit: is this 5000 some limit? Can we comment about it or assign to a variable and use?

File: desktop/libs/indexer/src/indexer/indexers/sql.py [hunk 4]
```diff
@@ -355,23 +357,30 @@ def create_table_from_local_file(self, source, destination, start_time=-1):
...
+            temp_list = _csv_rows[count*5000:(count+1)*5000]
+            if(len(temp_list)==0):
```
Comment (Harshg999) [inline desktop/libs/indexer/src/indexer/indexers/sql.py:364]: Nit: Remove ()

**Review by Harshg999 — COMMENTED** [review]
Looks good! Few nits

File: desktop/core/src/desktop/conf.py (hunk 1 above)
Comment (agl29) [reply desktop/core/src/desktop/conf.py]: Done

File: desktop/libs/indexer/src/indexer/indexers/sql.py (hunk 4 above)
Comment (agl29) [reply desktop/libs/indexer/src/indexer/indexers/sql.py]: Done

File: desktop/core/src/desktop/settings.py [hunk 5]
```diff
@@ -117,6 +118,8 @@
 # Examples: "http://media.lawrence.com/media/", "http://example.com/media/"
 MEDIA_URL = ''
 
+if ENABLE_DIRECT_UPLOAD:
+  DATA_UPLOAD_MAX_MEMORY_SIZE = 67108864  # Setting this varibale to 64MB as we are sending long POST requests
```
Comment (Harshg999) [inline desktop/core/src/desktop/settings.py:122]: Nit: Spelling error in comment.

**Review by Harshg999 — APPROVED** [review]
LGTM!

File: desktop/core/src/desktop/settings.py (hunk 5 above)
Comment (Harshg999) [reply desktop/core/src/desktop/settings.py:122]: `variable`

File: desktop/core/src/desktop/conf.py [hunk 6]
```diff
@@ -642,13 +642,16 @@ def default_from_email():
...
 
+from indexer.conf import ENABLE_DIRECT_UPLOAD  # Circular dependency 
 
 def default_database_options():
   """Database type dependent options"""
   if DATABASE.ENGINE.get().endswith('oracle'):
     return {'threaded': True}
   elif DATABASE.ENGINE.get().endswith('sqlite3'):
     return {'timeout': 30}
+  elif DATABASE.ENGINE.get().endswith('mysql') and ENABLE_DIRECT_UPLOAD:     # Setting this variale to 64MB because mysql5.7
```
Comment (Harshg999) [inline desktop/core/src/desktop/conf.py:653]: Here also `variable` @agl29

**Pull request merged**